sim/src/api.rs

```diff
@@ -35,3 +36,23 @@ fn map_err(err: Result<()>) -> libc::c_int {
         },
     }
 }
+
+/// Called by C code to determine if we should log at this level.  Levels are defined in
+/// bootutil/bootutil_log.h.  This makes the logging from the C code controlled by bootsim::api, so
+/// for example, it can be enabled with something like:
+///     RUST_LOG=bootsim::api=info cargo run --release runall
```

Comment:
Hmm, either works, without the ::api it applies to the whole sim, but with it, it applies to just the C code. I'll make them consistent.